packages/block-editor/src/components/block-pattern-setup/index.js

```diff
@@ -0,0 +1,187 @@
+ */
+import { useDispatch } from '@wordpress/data';
+import { cloneBlock } from '@wordpress/blocks';
+import {
+	VisuallyHidden,
+	__unstableComposite as Composite,
+	__unstableUseCompositeState as useCompositeState,
+	__unstableCompositeItem as CompositeItem,
+} from '@wordpress/components';
+
+import { useState } from '@wordpress/element';
+import { useInstanceId } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import BlockPreview from '../block-preview';
+import SetupToolbar from './setup-toolbar';
+import usePatternsSetup from './use-patterns-setup';
+import { VIEWMODES } from './constants';
+
+const SetupContent = ( {
+	viewMode,
+	activeSlide,
+	patterns,
+	onBlockPatternSelect,
+} ) => {
+	const composite = useCompositeState();
+	const containerClass = 'block-editor-block-pattern-setup__container';
+	let content;
+	if ( viewMode === VIEWMODES.carousel ) {
+		const slideClass = new Map( [
+			[ activeSlide, 'active-slide' ],
+			[ activeSlide - 1, 'previous-slide' ],
+			[ activeSlide + 1, 'next-slide' ],
+		] );
+		content = (
+			<div className={ containerClass }>
+				<ul className="carousel-container">
+					{ patterns.map( ( pattern, index ) => (
+						<BlockPatternSlide
+							className={ slideClass.get( index ) || '' }
+							key={ pattern.name }
+							pattern={ pattern }
+						/>
+					) ) }
+				</ul>
+			</div>
+		);
+	} else if ( viewMode === VIEWMODES.grid ) {
+		content = (
+			<Composite
+				{ ...composite }
+				role="listbox"
+				className={ containerClass }
+				aria-label={ __( 'Patterns list' ) }
+			>
+				{ patterns.map( ( pattern ) => (
+					<BlockPattern
+						key={ pattern.name }
+						pattern={ pattern }
+						onSelect={ onBlockPatternSelect }
+						composite={ composite }
+					/>
+				) ) }
+			</Composite>
+		);
+	}
+	return content;
+};
+
+function BlockPattern( { pattern, onSelect, composite } ) {
+	const baseClassName = 'block-editor-block-pattern-setup-list';
+	const { blocks, title, description, viewportWidth = 700 } = pattern;
+	const descriptionId = useInstanceId(
+		BlockPattern,
+		`${ baseClassName }__item-description`
+	);
+	return (
+		<div
+			className={ `${ baseClassName }__list-item` }
+			aria-label={ pattern.title }
+			aria-describedby={ pattern.description ? descriptionId : undefined }
+		>
+			<CompositeItem
+				role="option"
+				as="div"
+				{ ...composite }
+				className={ `${ baseClassName }__item` }
+				onClick={ () => onSelect( blocks ) }
+			>
+				<BlockPreview
+					blocks={ blocks }
+					viewportWidth={ viewportWidth }
+				/>
+				<div className={ `${ baseClassName }__item-title` }>
+					{ title }
+				</div>
+			</CompositeItem>
+			{ !! description && (
+				<VisuallyHidden id={ descriptionId }>
+					{ description }
+				</VisuallyHidden>
+			) }
+		</div>
+	);
+}
+
+function BlockPatternSlide( { className, pattern } ) {
```

Comment:
I think it's okay for now to have them in the same file, because some of them could probably be extracted and reused among various places. This is especially the case for `BlockPattern` component.